esmvaltool/diag_scripts/primavera/eady_growth_rate/eady_growth_rate.py

```diff
@@ -0,0 +1,210 @@
+import logging
+import os
+import sys
+
+import cartopy.crs as ccrs
+import iris
+import iris.analysis
+import iris.cube
+import iris.quickplot as qplt
+import iris.util
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+from dask import array as da
+
+import esmvaltool.diag_scripts.shared
+import esmvaltool.diag_scripts.shared.names as n
+from esmvalcore.preprocessor import (annual_statistics, extract_levels, regrid,
+                                     seasonal_statistics)
+from esmvaltool.diag_scripts.shared import ProvenanceLogger, group_metadata
+
+matplotlib.use('agg')
+
+logger = logging.getLogger(os.path.basename(__file__))
+
+
+class EadyGrowthRate(object):
+    def __init__(self, config):
+        self.cfg = config
+        self.filenames = esmvaltool.diag_scripts.shared.Datasets(self.cfg)
+        self.fill_value = 1e20
+        self.ref_p = 1000.0
+        self.g = 9.80665
+        self.con = 0.3098
+        self.omega = 7.292e-5
+        self.time_statistic = self.cfg['time_statistic']
+
+    def compute(self):
+        data = group_metadata(self.cfg['input_data'].values(), 'alias')
+        for alias in data:
+            var = group_metadata(data[alias], 'short_name')
+            ta = iris.load_cube(var['ta'][0]['filename'])
+            plev = ta.coord('air_pressure')
+
+            theta = self.potential_temperature(ta, plev)
+
+            del ta
+
+            zg = iris.load_cube(var['zg'][0]['filename'])
+
+            brunt = self.brunt_vaisala_frq(theta, zg)
+
+            lats = zg.coord('latitude')
+            fcor = self.coriolis(lats, zg.shape)
+
+            ua = iris.load_cube(var['ua'][0]['filename'])
+            if ua.shape is not zg.shape:
+                ua = regrid(ua, zg, scheme='linear')
+
+            egr = self.eady_growth_rate(fcor, ua, zg, brunt)
+
+            cube_egr = ua.copy(egr * 86400)
+
+            cube_egr.standard_name = None
+            cube_egr.long_name = 'eady_growth_rate'
+            cube_egr.var_name = 'egr'
+            cube_egr.units = 'day-1'
+
+            if self.time_statistic == 'annual_mean':
+                cube_egr = annual_statistics(cube_egr)
+                cube_egr = cube_egr.collapsed('time', iris.analysis.MEAN)
+            elif self.time_statistic == 'seasonal_mean':
+                cube_egr = seasonal_statistics(cube_egr)
+                cube_egr = cube_egr.collapsed('time', iris.analysis.MEAN)
+                self.seasonal_plots(cube_egr, alias)
+            else:
+                logger.info(
+                    "Parameter time_statistic is not well set in the recipe."
+                    "Must be 'annual_mean' or 'seasonal_mean'")
+                sys.exit()
+
+            self.save(cube_egr, alias, data)
+
+    def potential_temperature(self, ta, plev):
```

Comment:
Can you add a short docstring to describe what this does and what `ta` and `plev` arguments are?